### PR TITLE
Implement rand_s

### PR DESCRIPTION
**Pull request by thrimbor**
This implements the rand_s function, which is a MS-specific function required by libc++. To keep the executable size small and the code simple, it uses the kernel's SHA-1functions which are fed with the EEPROM contents, the TSC and the tick counter (the latter two are also used each time a new random number is requested).
The global state is initialized once by using `call_once` and protected by a mutex. Each time a random number is requested the hash is updated by feeding in the TSC and the tick counter, then the first four bytes of the output hash are extracted and returned as the random number, while also being fed back into the hash.

Security could of course be improved by adding additional entropy sources.

This PR conflicts with #185 as they touch the same file.

The CI failure is unrelated and described in #218.

### CODE REVIEW

File: lib/xboxrt/libc_extensions/stdlib_ext_.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    ULONG bytes_read;
+    ULONG eeprom_type;
+    ExQueryNonVolatileSetting(0xFFFF, &eeprom_type, eeprom_buffer, 256, &bytes_read);
+    assert(bytes_read == 256);
+    XcSHAUpdate(rand_buffer, eeprom_buffer, 256);
+
+    update_rand_buffer();
+}
+
+int rand_s (unsigned int *randomValue)
```
Comment (mborgerson) [inline lib/xboxrt/libc_extensions/stdlib_ext_.c]: May want to leave a comment that this probably isn't a very cryptographically secure random number generation as the MS docs indicate

File: lib/xboxrt/libc_extensions/stdlib_ext_.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#include <stdlib.h>
+#include <threads.h>
+#include <windows.h>
+#include <xboxkrnl/xboxkrnl.h>
+
+once_flag init_flag = ONCE_FLAG_INIT;
+mtx_t rand_mutex;
+unsigned char rand_buffer[116];
+
+void update_rand_buffer (void)
```
[the remark this answers is not in the document]
Comment (thrimbor) [reply lib/xboxrt/libc_extensions/stdlib_ext_.c]: I've been using this style for afaik everything I've written for winapi and xboxrt. I think it's a little easier on the eyes and differentiates declarations/definitions from calls.

File: lib/xboxrt/libc_extensions/stdlib_ext_.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+{
+    DWORD tickCount = GetTickCount();
+    XcSHAUpdate(rand_buffer, (PUCHAR)&tickCount, sizeof(tickCount));
+
+    LARGE_INTEGER performanceCounter;
```
[the remark this answers is not in the document]
Comment (thrimbor) [reply lib/xboxrt/libc_extensions/stdlib_ext_.c]: Yeah probably not. I removed the code using the tick count now.

File: lib/xboxrt/libc_extensions/stdlib_ext_.c (hunk 1 above)
Comment (thrimbor) [reply lib/xboxrt/libc_extensions/stdlib_ext_.c]: Done.

**Comment by JayFoxRox** [comment]
I feel this is a bit overkill; I strongly doubt anyone would attempt to attack Xbox (or its nxdk traffic / output) or the need for secure crypto in nxdk programs. I'd probably just have stubbed this and left a note about this being insecure.

If this is being implemented, I'd assume that SHA is problematic for performance; wouldn't RC4 make more sense? The drop-N variant should still be moderately secure.
Once initialized, it could just keep outputting bytes without a manual feedback loop.
SHA would only be used to initialize the RC4 state.

I don't care about this feature though, so I'll just ignore this PR. Someone else do final review / merge.

File: lib/xboxrt/libc_extensions/stdlib_ext_.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (JayFoxRox) [reply lib/xboxrt/libc_extensions/stdlib_ext_.c]: They likely aren't running exactly in sync due to time jitter.

I don't remember which counters are being used by each function - but even a single bit flip would change the hash a lot (as it's not just "adding" the counter values). So having as many different counters as possible would be good.

Then again: It probably doesn't matter for nxdk.

File: lib/xboxrt/libc_extensions/stdlib_ext_.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (JayFoxRox) [reply lib/xboxrt/libc_extensions/stdlib_ext_.c]: I think the variant without space is "easier on the eyes". I'm fine eitherway, as long as it is consistent within the file, and also the rest of the group it belongs to.

**Pull request merged**